improver/nbhood/recursive_filter.py

```diff
@@ -304,50 +304,56 @@ def _validate_smoothing_coefficients(cube, smoothing_coefficients_cube):
             ValueError: The coordinate to be smoothed within the
                 smoothing coefficient cube does not have the expected points.
         """
-        if smoothing_coefficients_cube.name() not in [
-            "smoothing_coefficient_x",
-            "smoothing_coefficient_y",
-        ]:
+        if smoothing_coefficients_cube.name() == "smoothing_coefficient_x":
+            smoothing_axis = "x"
+            non_smoothing_axis = "y"
+            smoothing_coord = smoothing_coefficients_cube.coord(axis="x")
+            non_smoothing_coord = smoothing_coefficients_cube.coord(axis="y")
+        elif smoothing_coefficients_cube.name() == "smoothing_coefficient_y":
+            smoothing_axis = "y"
+            non_smoothing_axis = "x"
+            smoothing_coord = smoothing_coefficients_cube.coord(axis="y")
+            non_smoothing_coord = smoothing_coefficients_cube.coord(axis="x")
+        else:
             msg = (
                 "The smoothing coefficients cube must be named either "
-                "smoothing_coefficient_x or smoothing_coefficient_y"
+                "smoothing_coefficient_x or smoothing_coefficient_y. "
+                "The smoothing coefficients are named: "
+                f"{smoothing_coefficients_cube.name()}"
             )
             raise ValueError(msg)
 
         mean_points = {
-            "x": (cube.coord(axis="x").points[1:] + cube.coord(axis="x").points[:-1])
-            / 2,
-            "y": (cube.coord(axis="y").points[1:] + cube.coord(axis="y").points[:-1])
-            / 2,
+            "x": (
+                (cube.coord(axis="x").points[1:] + cube.coord(axis="x").points[:-1]) / 2
+            ),
+            "y": (
+                (cube.coord(axis="y").points[1:] + cube.coord(axis="y").points[:-1]) / 2
+            ),
```

Comment:
We only use the mean_points on the smoothing_axis, therefore mean_points could be simplified to 
`mean_points = cube.coord(axis=smoothing_axis).points[1:] + cube.coord(axis=smoothing_axis).points[:-1]) / 2`